web/packages/teleport/src/Integrations/status/AwsOidc/Tasks/Tasks.tsx

```diff
@@ -0,0 +1,153 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useEffect, useState } from 'react';
+
+import { ButtonBorder, Flex, Indicator } from 'design';
+import { Danger } from 'design/Alert';
+import Table, { Cell } from 'design/DataTable';
+
+import { useServerSidePagination } from 'teleport/components/hooks';
+import { FeatureBox } from 'teleport/components/Layout';
+import { AwsOidcHeader } from 'teleport/Integrations/status/AwsOidc/AwsOidcHeader';
+import { AwsOidcTitle } from 'teleport/Integrations/status/AwsOidc/AwsOidcTitle';
+import { getResourceType } from 'teleport/Integrations/status/AwsOidc/helpers';
+import { TaskState } from 'teleport/Integrations/status/AwsOidc/Tasks/constants';
+import { Task } from 'teleport/Integrations/status/AwsOidc/Tasks/Task';
+import { useAwsOidcStatus } from 'teleport/Integrations/status/AwsOidc/useAwsOidcStatus';
+import { integrationService, UserTask } from 'teleport/services/integrations';
+
+export function Tasks() {
+  const { integrationAttempt } = useAwsOidcStatus();
+  const { data: integration } = integrationAttempt;
+  const [tasks, setTasks] = useState<UserTask[]>(undefined);
+  const [showTask, setShowTask] = useState<UserTask>(undefined);
+
+  const serverSidePagination = useServerSidePagination<UserTask>({
+    pageSize: 20,
+    fetchFunc: async () => {
+      const { items, nextKey } =
+        await integrationService.fetchIntegrationUserTasksList(
+          integration.name
+        );
+      setTasks(items);
+      return { agents: items, nextKey };
+    },
+    clusterId: '',
+    params: {},
+  });
+
+  useEffect(() => {
+    serverSidePagination.fetch();
+  }, [integration]);
+
+  if (integrationAttempt.status === 'processing') {
+    return <Indicator />;
+  }
+
+  if (integrationAttempt.status === 'error') {
+    return <Danger>{integrationAttempt.statusText}</Danger>;
+  }
+
+  function close(resolved: boolean) {
+    if (resolved) {
+      setTasks(p =>
+        p.map(item => {
+          if (item.name != showTask.name) {
+            return item;
+          }
+        })
+      );
+    }
+    setShowTask(undefined);
+  }
+
+  return (
+    <Flex>
+      <Flex
+        flexDirection="column"
+        css={`
+          flex-grow: 1;
+        `}
+      >
+        {integration && (
+          <AwsOidcHeader integration={integration} tasks={true} />
+        )}
+        <FeatureBox
+          css={{ maxWidth: '1400px', paddingTop: '16px', gap: '30px' }}
+        >
+          {integration && (
+            <AwsOidcTitle integration={integration} tasks={true} />
+          )}
+          <Table<UserTask>
+            data={tasks?.filter(t => t?.state === TaskState.Open) || []}
```

Comment:
If we change `.map` to `.filter` above.
```suggestion
            data={tasks?.filter(t => t.state === TaskState.Open) || []}
```

And just wanted to let you know that combining client-side filtering and server-side pagination can result in bugs. Above you used `serverSidePagination` which suggests that we want to use it to get next/prev page, show total count etc. But if we later filter the data on the client side, we may end up with a table showing total count 20 and a blank page because they have all been filtered out.

Is it possible to fetch only open tasks from the API?

Reply:
Excelllent point, following up with Marco.